streamlit_web.py

```diff
@@ -652,5 +657,38 @@ def interactive():
         st.write(summaries)
 
 
+def entities():
+    """Page to display entity analysis"""
+    st.header("Entity analysis inspects the given text for known entities \
+    and returns information about those entities. It is a way to extract \
+    information that seeks to locate and classify named entities in text \
+    into pre-defined categories such as the names of persons, organizations, \
+    locations, expressions of times, quantities, monetary values, and percentages.")
+    st.sidebar.success("Hello sidebar!")
+
+    # make a copy of the main dataframe
+    input_df = main_df.copy(deep=True)
+
+    # list out possible user documents to choose from and select one
+    students = st.multiselect(
+        label="Select specific students below:",
+        options=input_df[stu_id].unique(),
+    )
+
+    # create a dataframe with the selected user and convert it to a string
+    df_selected_stu = input_df.loc[input_df[stu_id].isin(students)]
+    student_string = df_selected_stu.to_string()
```

Comment:
Seems like this line is converting the entire data frame into string. This might include much information that is not necessary to display (tokens, escape symbols like `\n`). The headers are also not attached to the responses, making it also less readable.

An additional issue: when multiple assignments are selected in the sidebar, there is no separation or selection mechanism on the page. Currently, all the responses are just appended together. That's probably not the way we want it to function.

Reply:
@enpuyou I completely agree with you on the additional issue. One way of solving this can be done by creating an expandable section to hold the text for each student.
Here is an example of how that can look like and the code used to create it:
 
![Screenshot from 2021-04-11 13-43-44](https://user-images.githubusercontent.com/55197145/114315293-53828280-9acc-11eb-88ec-27de3a0aa704.png)

```python
import streamlit as st

def main():
    st.header("Example")
    data = {"student1" : "text1", "student2": "text2", "student3": "text3"}
    for name in data.keys():
        with st.beta_expander(name):
            st.write(data[name])


if __name__ == "__main__":
    main()
```

You can also take it a step further by creating columns but keep in mind that it might overcrowd the page. It would look like this:

![Screenshot from 2021-04-11 13-50-24](https://user-images.githubusercontent.com/55197145/114315410-e6bbb800-9acc-11eb-983e-554a6a25fa5d.png)

```python
import streamlit as st

def main():
    st.header("Example")
    data_1 = {"student1" : "text1", "student2": "text2", "student3": "text3"}
    data_2 = {"student4" : "text4", "student5": "text5", "student6": "text6"}
    col1, col2 = st.beta_columns(2)
    with col1:
        for name in data_1.keys():
            with st.beta_expander(name):
                st.write(data_1[name])
                
    with col2:
        for name in data_2.keys():
            with st.beta_expander(name):
                st.write(data_2[name])


if __name__ == "__main__":
    main()
```